new_log.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of the MAVLink Router project
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <errno.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <utility>
+
+#include "macro.h"
+
+class Log {
+public:
+    enum class Level {
+        ERROR = 0,
+        WARNING,
+        NOTICE,
+        INFO,
+        DEBUG,
+    };
+
+    static int open();
+    static int close();
+
+    static Level get_max_level() _pure_ { return _max_level; }
+    static void set_max_level(Level level);
+
+#define debug(...) log_internal(Log::Level::DEBUG, __VA_ARGS__)
+#define info(...) log_internal(Log::Level::INFO, __VA_ARGS__)
+#define notice(...) log_internal(Log::Level::NOTICE, __VA_ARGS__)
+#define warning(...) log_internal(Log::Level::WARNING, __VA_ARGS__)
+#define error(...) log_internal(Log::Level::ERROR, __VA_ARGS__)
```

Comment:
I did a template version that would work by means of std::forward... but then we lose the format attribute.